### DISCUSSION TITLE
Open Asset in "Edit" mode after Workflow transition happen

### DISCUSSION THREAD

**Discussion by alexmf23** [pimcore/pimcore#13809] [Q&A / Support]
<img width="1479" alt="Screenshot 2022-12-15 at 13 55 32" src="https://user-images.githubusercontent.com/63473281/207865121-21cd006f-b986-4f94-9d59-9198c48c588d.png">

Hello,
usually when you open an asset it will open in the "View" tab as default. I was looking for opening the asset in "Edit" tab when a specific workflow transition happen..

I can listen to the workflow.[workflow name].completed.[transition name] but how can i change the tab view ?

**Comment by mattamon** [comment]
Hey @alexmf23,

what you could basically try is to add some ExtJS script to check for your workflow in the asset.
https://github.com/pimcore/customer-data-framework/blob/3.3/src/Resources/public/js/startup.js#L322

There are asset events e.g. the preOpenAsset, where you could check the workflow state and set the view according to the state.
https://github.com/pimcore/pimcore/blob/10.5/bundles/AdminBundle/Resources/public/js/pimcore/events.js

You can implement something like the startup.js from the Customer Framework. You just have to register it in your AppBundle like so -> 
https://github.com/pimcore/customer-data-framework/blob/4.x/src/PimcoreCustomerManagementFrameworkBundle.php#L39

Be careful with the order of the files as it does create dependencies.
